fetch-in-react-practice/client/src/components/Todos.tsx

```diff
@@ -0,0 +1,153 @@
+import { useEffect, useState } from 'react';
+import PageTitle from './PageTitle';
+import TodoList from './TodoList';
+import TodoForm from './TodoForm';
+
+export type UnsavedTodo = {
+  task: string;
+  isCompleted: boolean;
+};
+export type Todo = UnsavedTodo & {
+  todoId: number;
+};
+
+export default function Todos() {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState<unknown>();
+
+  /* Implement useEffect to fetch all todos. Hints are at the bottom of the file. */
+  useEffect(() => {
+    async function fetchTodos() {
+      try {
+        const response = await fetch('/api/todos');
+        if (!response.ok) {
+          throw new Error(`Response invalid: ${response.status}`);
+        }
+        const fetchResponse = await response.json();
+        setTodos(fetchResponse);
+      } catch (error: any) {
```

Comment:
By default the type is `unknown`, which forces you to test its type before using it. The best way to write this is:

`console.error(error instanceof Error ? error.message : 'Unknown error');`

That way you test that it's an Error before grabbing its `message`.